openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -168,3 +171,90 @@ def cors_allow_xblock_handler(sender, request, **kwargs):  # lint-amnesty, pylin
+
+
+@view_auth_classes()
+class BlockFieldsView(APIView):
+    """
+    View to get/edit the field values of an XBlock as JSON (in the v2 runtime)
+
+    This class mimics the functionality of xblock_handler in block.py (for v1 xblocks), but for v2 xblocks.
+    However, it only implements the exact subset of functionality needed to support the v2 editors (from
+    the frontend-lib-content-components project). As such, it only supports GET and POST, and only the
+    POSTing of data/metadata fields.
+    """
+
+    @atomic
+    def get(self, request, usage_key_str):
+        """
+        retrieves the xblock, returning display_name, data, and metadata
+        """
+        try:
+            usage_key = UsageKey.from_string(usage_key_str)
+        except InvalidKeyError as e:
+            raise NotFound(invalid_not_found_fmt.format(usage_key=usage_key_str)) from e
+
+        block = load_block(usage_key, request.user)
+        block_dict = {
+            "display_name": get_block_display_name(block),
+            "data": block.data,
+            "metadata": block.get_explicitly_set_fields_by_scope(Scope.settings),
+        }
+        return Response(block_dict)
+
+    @atomic
+    def post(self, request, usage_key_str):
+        """
+        edits the xblock, saving changes to data and metadata only (display_name included in metadata)
+        """
+        try:
+            usage_key = UsageKey.from_string(usage_key_str)
+        except InvalidKeyError as e:
+            raise NotFound(invalid_not_found_fmt.format(usage_key=usage_key_str)) from e
+
+        user = request.user
+        block = load_block(usage_key, user)
+        data = request.data.get("data")
+        metadata = request.data.get("metadata")
+
+        old_metadata = block.get_explicitly_set_fields_by_scope(Scope.settings)
+        old_content = block.get_explicitly_set_fields_by_scope(Scope.content)
+
+        block.data = data
+
+        # update existing metadata with submitted metadata (which can be partial)
+        # IMPORTANT NOTE: if the client passed 'null' (None) for a piece of metadata that means 'remove it'.
+        if metadata is not None:
+            for metadata_key, value in metadata.items():
+                field = block.fields[metadata_key]
+
+                if value is None:
+                    field.delete_from(block)
+                else:
+                    try:
+                        value = field.from_json(value)
+                    except ValueError as verr:
+                        reason = _("Invalid data")
+                        if str(verr):
+                            reason = _("Invalid data ({details})").format(
+                                details=str(verr)
+                            )
+                        return JsonResponse({"error": reason}, 400)
+
+                    field.write_to(block, value)
+
+        if callable(getattr(block, "editor_saved", None)):
+            if old_metadata is None:
```

Comment:
`old_metadata` is set to the result of [`get_explicitly_set_fields_by_scope`](https://github.com/openedx/edx-platform/blob/57420ed6138add7e69f453ebc5ba8d5802b347b2/xmodule/x_module.py#L435), and it seems like that function never returns `None`. So is this `if` ever used? (same for `old_content`)

Reply:
fair enough. I was copying old v1 code from a function where those default to None if not passed in.